op-node/rollup/finality/finalizer.go

```diff
@@ -345,13 +347,22 @@ func tryFinalizeOnConsecutiveBlockQuorom(
 	// TODO: we shouldn't skip on no voting power.
 	// https://github.com/babylonchain/babylon-finality-gadget/issues/59
 	if err != nil && !errors.Is(err, sdkclient.ErrNoFpHasVotingPower) {
+		if lastFinalizedBlockNumber != nil {
+			// set finalized block(s)
+			finalizedL2Block := l2Blocks[*lastFinalizedBlockNumber]
+			fi.log.Debug("set finalized block", "l2_block", finalizedL2Block)
+			*finalizedL2 = finalizedL2Block
+			*finalizedDerivedFrom = fd.L1Block
+
+			return lastFinalizedBlockNumber, nil
```

Comment:
this is copied from below. we should avoid copying code as much as possible, sometimes even when there is one line of duplication.

this should be extracted to a helper function and call it here
```
return someFunc(...)
```

Reply:
wait, this logic seems wrong. seems we are suppressing the error here again?

Reply:
updated